MaxText/pyconfig.py

```diff
@@ -717,8 +717,8 @@ def calculate_global_batch_sizes(
 
 
 def get_num_target_devices(raw_keys):
-  compile_topology = accelerator_to_spec_map.get_system_characteristics(raw_keys.get("compile_topology", ""))
-  if compile_topology is not None:
+  if raw_keys["compile_topology"]:
```

Comment:
if its not in the map you'll get a KeyError exception rather than False. 
if "compile_topology" in raw_keys:

Reply:
Previously, len(jax.devices()) would be returned here in two cases -- "compile_topology" is not in raw_keys (in which case "" will be passed to get_system_characteristics) and compile_topology is in raw_keys but the value for it is not in our map of supported things.

Do you want to print the new error in both cases or just in one of them ?

Reply:
Whenever compile_topology is set (e.g. is not ""), we should error when its not a valid topology. Also we should only call get_system_characteristics when compile_toplogy is set

With AOT we set compile topology - we will get an error only when the compile_topology is not a valid a topology

Without AOT we don't set compile topology - we should return jax.devices()

Reply:
Thanks for explaining! In that case, what do you think about adding a comment above the if about AOT and doing either "if "compile_topology" in raw_keys:" or ".get" instead of [].

Reply:
Added a comment! Changed to using .get instead of []. Note that "compile_topology" in raw_keys would not work - every key such as compile_topology is part of raw_keys since they are defined in base.yml, but here we are really checking that it is not the empty string (which it is set to by default)